### PR TITLE
[ADD] Add partner_phone_number_extension module; add a phone number exte...

### PR DESCRIPTION
**Pull request by ehdem**
[ADD] Add partner_phone_number_extension module; add a phone number extension field in partner form view.

### CODE REVIEW

File: partner_phone_number_extension/res_partner_view.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    <record id="view_partner_number_extension_form" model="ir.ui.view">
+      <field name="name">view.partner.number.extension.form</field>
+      <field name="model">res.partner</field>
+      <field name="inherit_id" ref="base.view_partner_form"/>
+      <field name="arch" type="xml">
+        <field name="phone" position="after">
+           <field name="extension"/>
+        </field>
+        <xpath expr="//form[@string='Contact']/sheet/group/field[@name='phone']"
```
Comment (pedrobaeza) [inline partner_phone_number_extension/res_partner_view.xml]: You can use this abbreviated form for future maintenance:

```
<xpath expr="//form[@string='Contact']//field[@name='phone']"
```

Comment (ehdem) [inline partner_phone_number_extension/res_partner_view.xml]: tks

**Comment by hbrunn** [comment]
:+1:

**Comment by bwrsandman** [comment]
This module could be split into two in order to remove the dependency on base_contact
- partner_phone_extension -> base
- base_contact_phone_extension -> base_contact, partner_phone_extension

**Edit:** Done

**Comment by yvaucher** [comment]
:+1:

**Pull request merged**